seleniumJSEnv/src/main/scala/org/scalajs/jsenv/selenium/SeleniumRun.scala

```diff
@@ -171,18 +163,31 @@ private[selenium] object SeleniumRun {
   private def maybeCleanupDriver(d: WebDriver, config: SeleniumJSEnv.Config) =
     if (!config.keepAlive) d.close()
 
-  private def htmlPage(scripts: Seq[java.net.URL]): String = {
-    val scriptTags =
-      scripts.map(path => s"<script src='${path.toString}'></script>")
+  private def htmlPage(setupJsPath: Path, input: Seq[Input], materializer: FileMaterializer): String = {
+    val setupJs = makeTag(setupJsPath, "text/javascript", materializer)
+
+    val tags = input.map {
+      case Input.Script(path)   => makeTag(path, "text/javascript", materializer)
+      case Input.ESModule(path) => makeTag(path, "module", materializer)
+      case _                    => throw new UnsupportedInputException(input)
+    }
+
+    val allTags = setupJs +: tags
+
     s"""<html>
        |  <meta charset="UTF-8">
        |  <body>
-       |    ${scriptTags.mkString("\n    ")}
+       |    ${allTags.mkString("\n    ")}
        |  </body>
        |</html>
     """.stripMargin
   }
 
+  private def makeTag(path: Path, tpe: String, materializer: FileMaterializer): String = {
+    val url = materializer.materialize(path)
+    s"<script defer type='$tpe' src='${url.toString}'></script>"
```

Comment:
Also just noticed this: You can drop the `toString` (and hence the braces). It's implied by the `s` interpolator.